src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -2096,11 +2096,20 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 boundSecondArgWithConversions = MakeRValueAndIgnoreDiagnostics(boundSecondArg)
             End If
 
-            '  if there are still no errors check the original type of the first argument to be 
-            '       of a reference or nullable type, generate IllegalCondTypeInIIF otherwise
+            '  If there are still no errors check the original type of the first argument. First, we check
+            ' the pre-VB 16.0 condition, which is the first operand must be Nothing, a reference type, or
+            ' a nullable value type
             If Not hasErrors AndAlso Not (boundFirstArg.IsNothingLiteral OrElse boundFirstArg.Type.IsNullableType OrElse boundFirstArg.Type.IsReferenceType) Then
-                ReportDiagnostic(diagnostics, node.FirstExpression, ERRID.ERR_IllegalCondTypeInIIF)
-                hasErrors = True
+                ' VB 16 changed the requirements on the first operand to permit unconstrained type parameters. If we're in that scenario,
+                ' ensure that the feature is enabled and report an error if it is not
+                If Not boundFirstArg.Type.IsValueType AndAlso Not boundFirstArg.Type.IsNullableType() Then
```

Comment:
> AndAlso Not boundFirstArg.Type.IsNullableType() [](start = 53, length = 48)

Is this condition redundant?

Reply:
No. Unconstrained type parameters will hit this, and depending on the language version will bind successfully or not.

---
In reply to: [217213079](https://github.com/dotnet/roslyn/pull/29724#discussion_r217213079) [](ancestors = 217213079)